### PR TITLE
Test suite fixes when run with autoconfigured Coq

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: test-suite/coq-makefile/native4/run.sh [hunk 1]
```diff
@@ -31,6 +36,3 @@ sort > desired <<EOT
 ./test/.coq-native/Ntest_test.cmxs
```
Comment (SkySkimmer) [inline test-suite/coq-makefile/native4/run.sh:36]: @ppedrot IIUC this tests `-native-compiler no` in _CoqProject, so why do we expect coq-native files?

**Comment by Zimmi48** [comment]
The error in the Nix job seems related:

```
File "theories/.coq-native/Ntest_test.native", line 14, characters 3-40:
14 |   (NCoq_Init_Nat.const_NCoq_Init_Nat_add
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Unbound module NCoq_Init_Nat
Error: Native compiler exited with status 2 (in case of stack overflow,
       increasing stack size (typicaly with "ulimit -s") often helps)
make[4]: *** [Makefile:765: theories/test.vo] Error 1
make[3]: *** [Makefile:386: all] Error 2
make[3]: Leaving directory '/builds/coq/coq/nix-build-coq.drv-0/zpnm8nl7fx58rh4r7vmkdy384g810xw8-source/test-suite/coq-makefile/native1/_test'
==========> FAILURE <==========
    coq-makefile/native1/run.sh...Error!
```

**Comment by SkySkimmer** [comment]
I don't know how to debug the nix issue @coq/nix-maintainers

**Comment by Zimmi48** [comment]
IIUC these tests were being skipped since the Dune switch and thus were never really checked against Dune and coqnative?

IINM Nix is the only job that checks the test-suite after `make install`. Therefore, you *might* be able to reproduce the issue by doing `./configure -prefix ... && make && make install && make check` on your own machine...

**Comment by SkySkimmer** [comment]
>IIUC these tests were being skipped since the Dune switch and thus were never really checked against Dune and coqnative?

No, they were never skipped but they tested native compilation of a trivial file (with just a Qed proof in it) which is too trivial to detect bugs.
I changed the test to make it nontrivial, which required fixing the skip logic so that we don't get errors when native is configured off.

**Comment by SkySkimmer** [comment]
>IINM Nix is the only job that checks the test-suite after make install. Therefore, you might be able to reproduce the issue by doing ./configure -prefix ... && make && make install && make check on your own machine...

How does make check use the installed coq instead of the non-installed files?

**Comment by Zimmi48** [comment]
> How does make check use the installed coq instead of the non-installed files?

I'm not sure how things work.

**Comment by SkySkimmer** [comment]
I think the nix failure is because the default for native compile is `ondemand`. Let's try skipping also for ondemand.

**Comment by SkySkimmer** [comment]
That worked

File: test-suite/coq-makefile/native4/run.sh [hunk 2]
```diff
@@ -25,12 +20,5 @@ sort > desired <<EOT
 ./test/test_plugin.cmxs
 ./test/test.v
 ./test/test.vo
-./test/.coq-native
-./test/.coq-native/Ntest_test.cmi
-./test/.coq-native/Ntest_test.cmx
-./test/.coq-native/Ntest_test.cmxs
```
Comment (ejgallego) [inline test-suite/coq-makefile/native4/run.sh:31]: Umm , I'm lost as to what is going on with this test.

Comment (ejgallego) [reply test-suite/coq-makefile/native4/run.sh:31]: Wasn't it the point of the test to override the flag?

Comment (SkySkimmer) [reply test-suite/coq-makefile/native4/run.sh:31]: The `native4` test uses `-native-compiler no` (in the _coqproject) so no native products are expected.

Comment (ejgallego) [reply test-suite/coq-makefile/native4/run.sh:31]: Yup, but what about the above:
```
# Shoud override the _CoqProject flag "-native-compiler no"
export COQEXTRAFLAGS="-native-compiler yes"
```

Comment (SkySkimmer) [reply test-suite/coq-makefile/native4/run.sh:31]: I guess it doesn't
Let's change that in another pr though

Comment (ejgallego) [reply test-suite/coq-makefile/native4/run.sh:31]: I mean the above code was also removed, my question is why?

Comment (SkySkimmer) [reply test-suite/coq-makefile/native4/run.sh:31]: oh I didn't realize
These 4 near-identical tests were annoying to update so I made a mistake.

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**